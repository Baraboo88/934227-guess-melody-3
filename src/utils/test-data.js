export const welcomeScreenMockData = {
  numberOfErrors: 4,
  onStartButtonClick: () => {}
};

export const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [{
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `blues`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `jazz`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }],
  },
  {
    type: `artist`,
    song: {
      artist: `Jim Snow`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    },
    answers: [{
      picture: `https://api.adorable.io/avatars/128/0.6750575244607682`,
      artist: `John Daniels`,
    }, {
      picture: `https://api.adorable.io/avatars/128/0.4076094808626085`,
      artist: `Jack Beam`,
    }, {
      picture: `https://api.adorable.io/avatars/128/0.6972088175637985`,
      artist: `Jim Snow`,
    }],
  }
];

export const userAnswer = [false, false, false, false];
