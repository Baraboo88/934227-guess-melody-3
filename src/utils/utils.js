import {actionCreator} from "../reducer";

export const mapStateToProps = (state, props) => {
  if (state.step >= state.questions.length) {
    props.history.push(`/`);
  }
  const id = props.match.params.id;

  let question = state.questions.find((newOffer) => newOffer.id === Number(id));

  if (!question) {
    props.history.push(`/`);
  }
  const nextQuestion = state.questions[state.step + 1] ? {nextType: state.questions[state.step + 1].type, nextId: state.questions[state.step + 1].id} : {nextType: null, nextId: null};
  return {
    question,
    mistakes: state.mistakesNumber,
    nextType: nextQuestion.nextType,
    nextId: nextQuestion.nextId
  };
};

export const mapDispatchToProps = (dispatch) => ({
  onAnswer(question, answer) {
    dispatch(actionCreator.incrementMistake(question, answer));
    dispatch(actionCreator.incrementStep());
  }
});
