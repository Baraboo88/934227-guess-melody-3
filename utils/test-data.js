export const mockData = {
  numberOfErrors: 4
};
