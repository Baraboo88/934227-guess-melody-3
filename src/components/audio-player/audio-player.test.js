import renderer from 'react-test-renderer';
import React from 'react';
import AudioPlayer from './audio-player';
import {questions} from '../../utils/test-data';

it(`AudioPlayer successfully rendered`, () => {
  const tree = renderer
    .create(
        <AudioPlayer
          isPlaying={false}
          onPlayButtonClick={() => {}}
          src={questions[0].answers[0].src}
        />,
        {
          createNodeMock: () => {
            return {};
          }
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
