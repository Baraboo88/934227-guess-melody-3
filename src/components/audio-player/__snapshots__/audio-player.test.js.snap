// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer successfully rendered 1`] = `
Array [
  <button
    className="track__button track__button--play"
    data-test="test-play-button"
    disabled={true}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio />
  </div>,
]
`;
